From f5009c6554625c07beb46688d80548a7293e4627 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:42:10 +0800
Subject: [PATCH 45/60] GSI Android 17 compatibility:
 0047-Don-t-crash-system_server-if-we-fail-to-parse-dynami.patch

Change-Id: I0b03c5cc11b92fed5bc7d1a02a66e2cb3d5dcbc4
---
 .../fingerprint/aidl/FingerprintProvider.java  | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java b/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java
index 73a2af74f..bbc6aedf7 100644
--- a/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java
+++ b/services/core/java/com/android/server/biometrics/sensors/fingerprint/aidl/FingerprintProvider.java
@@ -266,13 +266,17 @@ public class FingerprintProvider implements IBinder.DeathRecipient, ServiceProvi
             android.util.Log.e("PHH-Enroll", " - " + loc.sensorLocationX + ", " + loc.sensorLocationY + ", " +loc.sensorRadius + ", disp =" + loc.display + ", shape " + loc.sensorShape);
         }
         if (prop.sensorLocations.length == 1 && prop.sensorLocations[0].sensorLocationX == 0) {
-            int[] otherValues = com.android.server.biometrics.AuthService.dynamicUdfpsProps(mContext);
-            if (otherValues.length > 0) {
-                SensorLocation loc = new SensorLocation();
-                loc.sensorLocationX = otherValues[0];
-                loc.sensorLocationY = otherValues[1];
-                loc.sensorRadius = otherValues[2];
-                prop.sensorLocations[0] = loc;
+            try {
+                int[] otherValues = com.android.server.biometrics.AuthService.dynamicUdfpsProps(mContext);
+                if (otherValues.length > 0) {
+                    SensorLocation loc = new SensorLocation();
+                    loc.sensorLocationX = otherValues[0];
+                    loc.sensorLocationY = otherValues[1];
+                    loc.sensorRadius = otherValues[2];
+                    prop.sensorLocations[0] = loc;
+                }
+            } catch(Throwable t) {
+                android.util.Log.e("PHH-Enroll", "FingerprintProvider dynamicUdfpsProps", t);
             }
         }
 
